src/common/row_operations/row_match.cpp

```diff
@@ -261,4 +261,124 @@ idx_t RowOperations::Match(DataChunk &columns, VectorData col_data[], const RowL
+	for (idx_t key_idx = 0; key_idx != keys_count; ++key_idx) {
+		auto key_type = condition_types[key_idx];
+		auto key_offset = col_offsets[key_idx];
+		switch (key_type.InternalType()) {
+		case PhysicalType::BOOL:
+		case PhysicalType::INT8:
+			TemplatedMatchRowsType<int8_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                   matches);
+			break;
+		case PhysicalType::INT16:
+			TemplatedMatchRowsType<int16_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                    matches);
+			break;
+		case PhysicalType::INT32:
+			TemplatedMatchRowsType<int32_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                    matches);
+			break;
+		case PhysicalType::INT64:
+			TemplatedMatchRowsType<int64_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                    matches);
+			break;
+		case PhysicalType::UINT8:
+			TemplatedMatchRowsType<uint8_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                    matches);
+			break;
+		case PhysicalType::UINT16:
+			TemplatedMatchRowsType<uint16_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                     matches);
+			break;
+		case PhysicalType::UINT32:
+			TemplatedMatchRowsType<uint32_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                     matches);
+			break;
+		case PhysicalType::UINT64:
+			TemplatedMatchRowsType<uint64_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                     matches);
+			break;
+		case PhysicalType::INT128:
+			TemplatedMatchRowsType<hugeint_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                      matches);
+			break;
+		case PhysicalType::FLOAT:
+			TemplatedMatchRowsType<float, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                  matches);
+			break;
+		case PhysicalType::DOUBLE:
+			TemplatedMatchRowsType<double, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                   matches);
+			break;
+		case PhysicalType::INTERVAL:
+			TemplatedMatchRowsType<interval_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                       matches);
+			break;
+		case PhysicalType::VARCHAR:
+			TemplatedMatchRowsType<string_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
+			                                     matches);
+			break;
+		case PhysicalType::LIST:
+		case PhysicalType::MAP:
+		case PhysicalType::STRUCT:
+			matches = 1; // not handling nested types because it requires multiples passes over the hashtable
```

Comment:
Could we throw an exception in this case, and handle the bail-out earlier on and more explicitly (e.g. by adding a type switch in the join HT that initializes `has_unique_keys` to false if any unsupported types are found)?